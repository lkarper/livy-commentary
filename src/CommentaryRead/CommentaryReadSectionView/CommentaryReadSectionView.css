.CommentaryReadSectionView__container {
    display: flex;
    flex-direction: row;
    width: 100%;
}

.CommentaryReadSectionView__section {
    width: 50%;
    margin: 5px;
    padding: 10px;
    border: 1px solid black;
    word-wrap: break-word;
}