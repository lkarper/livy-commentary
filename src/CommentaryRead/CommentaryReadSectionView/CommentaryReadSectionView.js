import React from 'react';
import CommentaryReadComment from '../CommentaryReadComment/CommentaryReadComment';
import './CommentaryReadSectionView.css';

const CommentaryReadSectionView = (props) => {

    const { section } = props;

    return (
        <div className='CommentaryReadSectionView__container'>
            <section className='CommentaryReadSectionView__section'>
                <h3>{section.section_number}</h3>
                <p>{section.latin}</p>
            </section>
            <section className='CommentaryReadSectionView__section'>
                <h3>Commentary and Notes</h3>
                {section.comments.length !==0 && 
                    section.comments
                        .sort((a, b) => a.comment_order - b.comment_order)
                        .map((comment, i) => <CommentaryReadComment key={i} comment={comment} />)}
                {section.comments.length === 0 && <p>No notes yet for this section.</p>}
            </section>
        </div>
    );
}

export default CommentaryReadSectionView;