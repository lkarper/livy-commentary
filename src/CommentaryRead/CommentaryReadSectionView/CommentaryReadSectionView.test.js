import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CommentaryReadSectionView from './CommentaryReadSectionView';

describe('CommentaryReadSectionView component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <CommentaryReadSectionView />,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<CommentaryReadSectionView />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
