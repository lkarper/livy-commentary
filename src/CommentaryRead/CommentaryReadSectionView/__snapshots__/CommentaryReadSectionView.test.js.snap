// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentaryReadSectionView component renders the UI as expected 1`] = `
<CommentaryReadSectionView
  section={
    Object {
      "comments": Array [],
      "latin": "",
      "section_number": "",
    }
  }
>
  <div
    className="CommentaryReadSectionView__container"
  >
    <h3>
      Error
    </h3>
    <p>
      Looks like something went wrong; check your connection and try again.
    </p>
  </div>
</CommentaryReadSectionView>
`;
